Fix potential exception when the reference in InputActionProperty is null

chris-massie opened the pull request:
This PR updates `InputActionProperty` to handle when the reference is `null`.

Tested with a behavior in the XR Interaction Toolkit that serialized an `InputActionProperty` and used the inspector to delete the reference while keeping **Use Reference**=`true`, and accessing the `InputActionProperty.action` property no longer threw.

chris-massie committed:
Fix exception when the reference is null in InputActionProperty

Rene-Damm commented on Packages/com.unity.inputsystem/InputSystem/Actions/InputActionProperty.cs:
```diff
@@ -33,12 +33,12 @@ public struct InputActionProperty : IEquatable<InputActionProperty>, IEquatable<
         /// <see cref="InputActionProperty(InputAction)"/>.
         /// </remarks>
-        public InputAction action => m_UseReference ? m_Reference.action : m_Action;
+        public InputAction action => m_UseReference ? m_Reference != null ? m_Reference.action : null : m_Action;
```
`m_UseReference ? m_Reference?.action : m_Action`

chris-massie replied:
I would have used `?.` but that wouldn't use the overloaded `operator==` of `UnityEngine.Object` which also checks if the object is valid.

Rene-Damm reviewed (commented):
Needs test.

chris-massie committed:
Add tests for InputActionProperty and fix throw in GetHashCode

- Added tests
- Fixed NullReferenceException in GetHashCode for null actions

chris-massie commented:
I added tests including one that would fail if the property was changed to `public InputAction action => m_UseReference ? m_Reference?.action : m_Action;`

I also fixed the `NullReferenceException` in `GetHashCode` for a `null` `InputAction`.

Commit:
MERGE: develop => fix-inputactionproperty-exception.
